tests/composition/test_autodiffcomposition.py

```diff
@@ -454,7 +454,7 @@ def test_semantic_net_training_correctness(self, eps, opt, autodiff_mode, benchm
 
         for res, exp in zip(results, expected):
             for r, e in zip(res, exp):
-                assert np.allclose(r, e)
```

Comment:
It should be, it looked correct locally for me so I'll double check this after updating with current devel.

This loop+zips can't be replaced with a single command in this case I think because results and expected have ragged arrays in them, and this causes it to fail with `TypeError: ufunc 'isfinite' not supported for the input types, and the inputs could not be safely coerced to any supported types according to the casting rule ''safe''`